src/server/search/doc_accessors.h

```diff
@@ -28,14 +28,25 @@ struct BaseAccessor : public search::DocumentAccessor {
   virtual SearchDocData Serialize(const search::Schema& schema) const = 0;
 
   // Serialize selected fields
-  virtual SearchDocData Serialize(const search::Schema& schema, const FieldsList& fields) const;
+  virtual SearchDocData Serialize(const search::Schema& schema,
+                                  const SearchFieldsSvList& fields) const;
 
   /*
   Serialize the whole type, the default implementation is to serialize all fields.
   For JSON in FT.SEARCH we need to get answer as {"$", <the whole document>}, but it is not an
   indexed field
   */
   virtual SearchDocData SerializeDocument(const search::Schema& schema) const;
+
+  // Serialize selected fields
+  SearchDocData Serialize(const search::Schema& schema, const SearchFieldsList& fields) const {
+    SearchFieldsSvList sv_fields;
+    sv_fields.reserve(fields.size());
+    for (const auto& [ident, name] : fields) {
+      sv_fields.emplace_back(ident, name);
+    }
+    return Serialize(schema, sv_fields);
+  }
```

Comment:
Not sure why you suddently need both string and string_views 🤷🏻‍♂️ If you really do, you can avoid duplicating the interface by using base::it::Wrap

```c++

template<typename S>
using FieldSpan = absl::Span<const std::pair<S, S>>;

void Serialize(variant<FieldSpan<string>, FieldSpan<string_view>> fields) {
  auto mapper = [](const auto& p) -> pair<string_view, string_view> { return {{p.first}, {p.second}}; };
  for (auto [name, ident] : base::it::Wrap(mapper, fields))
    ...
}

```